data_structures/src/chain.rs

```diff
@@ -2578,6 +2579,22 @@ mod tests {
         assert!(a < b);
     }
 
+    #[test]
+    fn hash_from_str() {
+        // 32 bytes
+        let a = Hash::from_str("1111111111111111111111111111111111111111111111111111111111111111");
+        assert!(a.is_ok());
+
+        // 32.5 bytes
+        let b = Hash::from_str("11111111111111111111111111111111111111111111111111111111111111112");
+        assert!(b.is_err());
```

Comment:
If we do decide to use `FromHexError` directly, then we can even assert equality like so:

```suggestion
        assert_eq!(b, Err(FromHexError::InvalidStringLength));
```